src/shared/json.h

```diff
@@ -259,6 +263,59 @@ enum {
 #define JSON_BUILD_HEX(p, n) _JSON_BUILD_HEX, (const void*) { p }, (size_t) { n }
 #define JSON_BUILD_ID128(id) _JSON_BUILD_ID128, (const sd_id128_t*) { &(id) }
 #define JSON_BUILD_BYTE_ARRAY(v, n) _JSON_BUILD_BYTE_ARRAY, (const void*) { v }, (size_t) { n }
+#define JSON_BUILD_IN4_ADDR(v) JSON_BUILD_BYTE_ARRAY(v, sizeof(struct in_addr))
```

Comment:
hmm, so the macros above try hard to be typesafe. i.e. if you pass wrongly typed stuff the compiler will refuse. To make that work we use compound literals everywhere, even for simple types. please do the same here, i.e. define:

```c
JSON_BUILD_IN4_ADDR(v) JSON_BUILD_BYTE_ARRAY((const struct in_addr*) { v }, sizeof(struct in_addr))
```

That way if you then use `JSON_BUILD_IN4_ADDR("foo")` this will generate a compiler error, where it previously did not.

(and similar for the other types)